### PR TITLE
Fixes #24923 - Add check for duplicate puppet classes

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by iNecas** [comment]
Rubocop is not happy.

**Comment by ares** [comment]
Rubocop now jumps up and down overdosed with happiness

File: definitions/checks/foreman/puppet_class_duplicates.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  module Foreman
+    class PuppetClassDuplicates < ForemanMaintain::Check
+      metadata do
+        label :puppet_class_duplicates
+        for_feature :foreman_database
+        description 'Check for duplicate Puppet class records'
+        tags :pre_upgrade
+        confine do
+          feature(:foreman_database) &&
+            feature(:foreman_database).check_max_version('foreman', '1.20')
```
Comment (iNecas) [inline definitions/checks/foreman/puppet_class_duplicates.rb]: actually, this confine block could be eliminated to:

```
confine do
  check_max_version('foreman', '1.20')
end
```

Comment (iNecas) [reply definitions/checks/foreman/puppet_class_duplicates.rb]: The `for_feature` already implies that the feature is there, and the `check_max_version` should be available in this context already

**Comment by ares** [comment]
updated

**Comment by iNecas** [comment]
Tested with:

```
$ for x in {1,2,3}; do sudo su postgres - -c 'psql -d foreman -c "INSERT INTO puppetclasses (name) VALUES ('\''myclass'\'');"'; done

$ foreman-maintain health check --label puppet_class_duplicates
Check for duplicate Puppet class records:                             [FAIL]
There are 1 Puppet classes with duplicities:
myclass - 6
Please head over to Configure -> Classes and make sure there is only 1 Puppet class record for each name.

$ sudo su postgres - -c 'psql -d foreman -c "DELETE FROM puppetclasses WHERE name = ('\''myclass'\'');"';
# sudo su postgres - -c 'psql -d foreman -c "INSERT INTO puppetclasses (name) VALUES ('\''myclass'\'');"'

$ foreman-maintain health check --label puppet_class_duplicates
Running ForemanMaintain::Scenario::FilteredScenario
================================================================================
Check for duplicate Puppet class records:                             [OK]

```

**Pull request merged**